CMakeLists.txt

```diff
@@ -6,7 +6,10 @@
 
 cmake_minimum_required(VERSION 3.7.2)
 
-include(settings.cmake)
+if(NOT SEL4TEST_DONT_USE_SEL4_DEFAULT_BUILD_SYSTEM)
+    # include the settings from the standard sel4 build system
+    include(settings.cmake)
```

Comment:
What if you added a variable `SEL4TEST_SETTINGS_CONFIGURED` and used it like an include guard at the top of settings.cmake
```
get_property(include_guard GLOBAL PROPERTY SEL4TEST_SETTINGS_CONFIGURED)
if (include_guard)
    return()
endif()
set_property(GLOBAL PROPERTY SEL4TEST_SETTINGS_CONFIGURED "ON")
```

Reply:
What I consider a bit ugly about this is, that all I really want to is not including `settings.cmake`. Clearly doing this in the `CMakeLists.txt` seems the cleanest thing. ... but actually, having some parts of `settings.cmake` would be nice because there are project specific settings there.I have to try this out and see where the real (build system) conflicts are with this file.

Reply:
Do you still need to not include settings.cmake?  It's now safe to call sel4_configure_platform_settings() multiple times which it may not have been when this PR was originally created.